src/SurfaceFluxes.jl

```diff
@@ -365,74 +360,66 @@ function obukhov_length(
+        function root_l_mo(x_lmo)
+            residual = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
+            return residual
         end
-        return sol, L_MO, is_converged
-    end
-    sol, L_MO, is_converged = solve_lmo(ΔDSEᵥ)
-    if sol != nothing && !sol.converged
-        KA.@print("-----------------------------------------\n")
-        KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
-        KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
-        KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
-        KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
-        KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
-        KA.@print(", u_in = ", u_in(sc))
-        KA.@print(", u_sfc = ", u_sfc(sc))
-        KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
-        KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
-        KA.@print(", Δz = ", Δz(sc))
-        KA.@print(", ΔDSE = ", ΔDSE)
-        if error_on_non_convergence()
-            error("Unconverged Surface Fluxes.")
-        else
-            KA.@print("Warning: Unconverged Surface Fluxes. Returning iteration history.")
-            if soltype isa CompactSolution
-                @show sol.root
+        # Return the iterated solution for Monin-Obukhov length in non-neutral boundary layer case
+        sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), soltype, tol, maxiter)
+        L_MO = sol.root
+        if !sol.converged
+            if error_on_non_convergence()
+                KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
+                KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
+                KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
+                KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
+                KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
+                KA.@print(", u_in = ", u_in(sc))
+                KA.@print(", u_sfc = ", u_sfc(sc))
+                KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
+                KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
+                KA.@print(", Δz = ", Δz(sc))
+                KA.@print(", ΔDSEᵥ = ", ΔDSEᵥ)
+                if soltype isa RS.CompactSolution
+                    KA.@print(", sol.root = ", sol.root)
+                else
+                    KA.@print(", sol.root_history = ", sol.root_history)
+                    KA.@print(", sol.err_history = ", sol.err_history)
+                end
+                error("Unconverged Surface Fluxes.")
             else
-                @show sol.err_history
-                @show sol.root_history
-                KA.@print("-----------------------------------------\n")
+                KA.@print("Warning: Unconverged Surface Fluxes. Returning last interation.")
```

Comment:
We should probably remove this. What I've seen happen when doing this is that jobs appear to hang due to printing enormous logs

Reply:
So what you suggest is that we either break the simulation and print the information, or continue the simulation without printing anything?

Reply:
Yes, I suggest printing the error info and erroring out. We probably shouldn't ever run with `error_on_non_convergence() = false`. I believe that this was added to mirror thermodynamics, which is useful for writing JET tests. We'll likely need this here, too.

Reply:
Ok, It just would be nice to have an option to not erroring out when it doesn't converge for now. But we can remove it.